pippy/PipelineSchedule.py

```diff
@@ -314,9 +336,31 @@ def step_microbatches(
         for work in bwd_sends_to_wait:
             work.wait()
 
-    def step(self, *args, **kwargs):
-        # TODO
-        pass
+        # Return losses if there is a container passed in
+        if losses is not None:
+            # Clean external container first
+            losses.clear()
```

Comment:
wondering if its better practice to request users clear or provide empty container, vs erasing it for them.  i could see this is convenient sometimes.

Reply:
Here is more of a safety thinking. I don't want a case where `len(losses)` ends up not being equal to `n_microbatches`, as I am using `extend()` below.